controllers/phish.go

```diff
@@ -126,6 +127,32 @@ func (ps *PhishingServer) registerRoutes() {
 	ps.server.Handler = phishHandler
 }
 
+// ArbitraryEventHandler deals with arbitrary events - for example opening Word documents, secondary links, etc
+func (ps *PhishingServer) ArbitraryEventHandler(w http.ResponseWriter, r *http.Request) {
+
+	r, err := setupContext(r)
+	if err != nil {
+		// Log the error if it wasn't something we can safely ignore
+		if err != ErrInvalidRequest && err != ErrCampaignComplete {
+			log.Error(err)
+		}
+		http.NotFound(w, r)
+		return
+	}
+
+	rs := ctx.Get(r, "result").(models.Result)
+	d := ctx.Get(r, "details").(models.EventDetails)
+
+	err = rs.HandleArbitraryEvent(d)
+	if err != nil {
+		log.Error(err)
+		w.Write([]byte(err.Error()))
+	} else {
+
```

Comment:
Small nit - blank line here.

As a side note, I'd recommend setting up `goimports` to run as part of your editor on save. That way, `gofmt` is automatically run and imports are automatically resolved.

Reply:
Hmm I do have goimports with my Visual Studio Code, but it's been acting weirdly lately - I noticed this too on my last pull request. Doing some Googling it seems to be a known issue with OSX + VS + Go. I'll try fix this.